mdf.md

```diff
@@ -0,0 +1,83 @@
+#  MVDS Metadata Field
+
+> Version: 0.1.0 (Draft)
+> 
+> Authors: Oskar Thorén <oskar@status.im>, Dean Eigenmann <dean@status.im>
+
+##  Table of Contents
+
+1. [Abstract](#abstract)
+2. [Motivation](#motivation)
+3. [Format](#format)
+    1. [Fields](#fields)
+4. [Usage](#usage)
+    1. [Informational Fields](#informational-fields)
+    2. [Configurational Fields](#configurational-fields)
+
+## Abstract
+
+In this specification, we describe a method to construct message history that will aid the consistency of [MVDS](./mvds.md), which currently do not exist, the additional fields allow for various consistency models such. Additionally we explain how data sync can be used for more lightweight messages that do not require full synchronization.
+
+## Motivation
+
+In order for more efficient synchronization of conversational messages, information should be provided allowing a node to more effectively synchronize the dependencies for any given message. Encoding a DAG within a message allows for an entire portion to be synchronized if messages are missing with a single `REQUEST` message.
+
+## Format
+
+We introduce the metadata message which is used to convey information about a message and how it SHOULD be handled. If an MVDS node implements this, we say that node has MDF capability (This might become part of a future reconciled MVDS spec).
+
+
+```protobuf
+package vac.mvds;
+
+message Metadata {
```

Comment:
What happened with the sequence field?

Reply:
We decided to temporarily remove, due to it potentially not being as useful as previously expected.

Reply:
I am actually fine keeping seq number if we want to do so, to me the useful fields are (naming can change, just making it verbose so we understand what we mean):

1) `Local parent`:
What?  The previous message that this node has sent in this group ID 
Why? Useful if we want to traverse a single node dependency chain
Use case: Traversing through a remote log where the location is known and not encoded in the hash (i.e I know where to find messages from X, and I can query by MessageID, but given MessageID and not knowing X I can't find the message)
What does it provide: Linearizability, in the context of {peerID, messageID}, trivially achieved

2) `Global parents`
What? The N parents (not ancestors) of this message. N is bound by [0, number-of-peers-in-the-group].
Why? Useful to traverse the dependency graph
Use Case: Just pulling the history of a message
What does it provide: eventual (causal) consistency if messages are passed to the client immediately, causal consistency if messages are not passed to the client until dependencies are satisfied. Given that you have full causal history you can have a partial order, so only in causal dependency mode this is useful, while in eventual consistency mode gaps won't allow you to properly causally order events).

What would be also useful is to have a logical clock (lamport timestamp), for ordering of events when gaps occur, useful in eventual consistency model or in causal consistency model when the dependency is using sparsely.

Reply:
As for sequence number: let's add as a comment to consider it for future versions and be done with it. Easy to add if there's a need. Let's let specific use cases drive this, e.g. dasy, status app, ulb, libp2p etc

Reply:
> To me local parent is what we was previously called previous in this PR. Do you disagree with this? Not sure why we removed it.

@oskarth we removed it because it can be summarized in the `parents` field. I do not think that we really need to differentiate between local and global at this layer. If it becomes useful later on we can always readd the field. It's easy to add fields.

Reply:
Another thing we should probably specify a bit better is what a "parent" is, as it might not be immediately intuitive. 
Something on the lines: 

A parent of a new message is any node in that DAG that the client is aware that has no children, the number of parents is bound by [0, N], where N is the number of peers participating in the conversation, therefore the space requirements are O(N). If a message has no  parents it is considered a root. There can be multiple roots, which might be disconnected, giving rise to multiple DAGs.

Reply:
Perfect, thanks @cammellos